### ISSUE TITLE
Support shortcuts for chats

### ISSUE THREAD

**Issue by roihershberg** [evilcorpltd/aTox#944]
Support adding shortcuts for chats in the phone's launcher.

**Comment by robinlinden** [comment]
This is part of #651.

**Comment by roihershberg** [comment]
Does the issue you referenced will support 4 shortcut chats that will be displayed when long-pressing the app's icon in the launcher as well as creating a shortcut from the widgets screen in which it will launch a special activity for choosing the desired contact as well as a menu item like "Add to homescreen" from the chat fragment?
That's at least what I'm planning to implement. Or you meant that some of the functionality of what I described also exists in the bubbles implementation?

**Comment by robinlinden** [comment]
>Does the issue you referenced will support 4 shortcut chats that will be displayed when long-pressing the app's icon in the launcher as well as creating a shortcut from the widgets screen in which it will launch a special activity for choosing the desired contact as well as a menu item [...]

I've implemented shortcuts bringing you straight to chats with specific contacts that appear when long-pressing the app's icon. It's a requirement for chat bubbles.

**Comment by roihershberg** [comment]
Ah nice! Then maybe after the bubbles we can expand the functionality to a special activity that will be launched by dragging and dropping a 1x1 widget from the classic Android Widgets section and it will allow you to choose any contact you want to create a shortcut for. Also a "Add to homescreen" menu item in chat fragment would be nice.

**Comment by robinlinden** [comment]
Yeah, I guess this can be expanded a bit. :P I've only set up the ShortcutManager things I needed for the chat bubbles to work, so the widget and shortcuts in the app can be done in this issue.

**Comment by robinlinden** [comment]
I'll finish the ShortcutManager bits of the chat bubbles and unassign myself from this issue once they're in, and you can add the other bits you wanted if you want!

**Comment by roihershberg** [comment]
Nice!
